frontend/src/modules/entitydetails/components/EntityDetails.js

```diff
@@ -92,18 +92,28 @@ export class EntityDetailsBase extends React.Component<InternalProps, State> {
         }
     }
 
+    /*
+     * Only fetch helpers data if the entity has changed.
+     */
     fetchHelpersData() {
         const { dispatch, locale, parameters, pluralForm, selectedEntity } = this.props;
 
         if (!parameters.entity || !selectedEntity || !locale) {
             return;
         }
 
-        dispatch(history.actions.get(parameters.entity, parameters.locale, pluralForm));
-        dispatch(otherlocales.actions.get(parameters.entity, parameters.locale));
+        if (selectedEntity.pk !== this.props.history.entity) {
+            dispatch(history.actions.get(parameters.entity, parameters.locale, pluralForm));
+        }
+
+        if (selectedEntity.pk !== this.props.otherlocales.entity) {
+            dispatch(otherlocales.actions.get(parameters.entity, parameters.locale));
+        }
 
-        const source = utils.getOptimizedContent(selectedEntity.original, selectedEntity.format);
-        dispatch(machinery.actions.get(source, locale, selectedEntity.pk));
+        if (selectedEntity.pk !== this.props.machinery.entity) {
+            const source = utils.getOptimizedContent(selectedEntity.original, selectedEntity.format);
+            dispatch(machinery.actions.get(source, locale, selectedEntity.pk));
+        }
```

Comment:
Or maybe I could just add one big `if` and return in case any of the helpers has old data? It's unlikely they will have data for different entities anyways.

Reply:
Yeah, especially now that I added the `pluralForm` check, it makes no sense to make that change.